app/src/main/java/com/onboarding/parkingsystemkotlin/database/ParkingDatabase.kt

```diff
@@ -22,7 +23,32 @@ object ParkingDatabase {
 
     fun getParkingLots() = parkingLots
 
-    fun setParkingLots(lots: Int){
+    fun setParkingLots(lots: Int) {
         parkingLots = lots
     }
+
+    fun removeOldReservations(): Int {
+        val currentTime = GregorianCalendar()
+        var deleted: Int = ConstantUtils.ZERO
+        reservations.forEach { (lot, list) ->
+            run {
+                deleted += searchForOldReservation(currentTime, reservations[lot])
+                if (list.isNullOrEmpty()) {
+                    reservations.remove(lot)
+                }
+            }
+        }
+        return deleted
+    }
+
+    private fun searchForOldReservation(current: GregorianCalendar, reservations: ArrayList<Reservation>?): Int {
+        var deletedCurrentLot = ConstantUtils.ZERO
+        reservations?.forEach {
+            if (current.after(it.getEndDate())) {
+                reservations.remove(it)
+                deletedCurrentLot++
+            }
+        }
+        return deletedCurrentLot
```

Comment:
Intead to do that forEach to reservations, you can use a filter so the condition that you're going to check inside is the same that you have in line 47.
return reservations?.filter { condition }.length  -> There would be something like this

Reply:
done